ARMmbed/theia
# Vscode telemetry

#### What it does

- adds code for vscode extension telemetry from later theia version (https://github.com/eclipse-theia/theia/pull/12453)
- updates that code by enabling telemetry and adding extension name as a prefix to the event names (like vscode does)

#### How to test

- checkout and and build locally
- map the cmsis-csolution extension to a checked out version on this branch: https://github.com/Arm-Debug/vscode-cmsis-csolution/tree/bump-telemetry
- in local KSC, switch between a cmsis extension to another one
- in the Azure dashboard, track the 'arm.cmsis-csolution/solutionLoaded' event for last 30 mins. The number should go up after a few switches of projects

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
